File: 2546-NumberOfSubarraysWithGcdEqualToK/2546-NumberOfSubarraysWithGcdEqualToK.java
// Last updated: 8/13/2025, 8:15:56 PM
class Solution {
    public int subarrayGCD(int[] nums, int k) {
        int count = 0;
        int n = nums.length;
        for(int i=0;i<n;i++){
            int ans = 0;
            for(int j=i;j<n;j++){
                ans = gcd(nums[j],ans);
                if(ans<k){
                    break;
                }
                if(ans==k){
                    count++;
                }
            }
        }
        return count;
    }

    public int gcd(int a , int b){
        if(a==0){
            return b;
        }
        int temp = gcd(b%a,a);
        return temp;
    }
}

File: 2546-NumberOfSubarraysWithGcdEqualToK/2546-NumberOfSubarraysWithGcdEqualToK.py
# Last updated: 8/13/2025, 8:15:50 PM
class Solution:
    def subarrayGCD(self, nums: List[int], k: int) -> int:
        count = 0
        for i in range(len(nums)) : 
            ans = 0
            for j in range (i , len(nums)) : 
                ans = self.gcd(ans , nums[j])
                if ans < k :
                    break
                if ans == k : 
                    count+=1
        return count
    
    def gcd(self , a : int , b : int) : 
        if a == 0 : 
            return b
        
        temp = self.gcd(b%a , a) 
        return temp